python/ray/services.py

```diff
@@ -356,6 +356,143 @@ def check_version_info(redis_client):
         else:
             print(error_message)
 
+def start_credis_component(module="member",
+                           node_ip_address="127.0.0.1",
+                           port=None,
+                           num_retries=20,
+                           stdout_file=None,
+                           stderr_file=None,
+                           cleanup=True):
+    """Start a single credis server with a given module loaded.
+
+    Args:
+        module (str): The credis module to be loaded, can be "master" or
+            "member".
+        node_ip_address (str): The IP address of the current node. This is only
+            used for recording the log filenames in Redis.
+        port (int): If provided, start a Redis server with this port.
+        num_retries (int): The number of times to attempt to start Redis. If a
+            port is provided, this defaults to 1.
+        stdout_file: A file handle opened for writing to redirect stdout to. If
+            no redirection should happen, then this should be None.
+        stderr_file: A file handle opened for writing to redirect stderr to. If
+            no redirection should happen, then this should be None.
+        cleanup (bool): True if using Ray in local mode. If cleanup is true,
+            then this process will be killed by serices.cleanup() when the
+            Python process that imported services exits.
+
+    Returns:
+        A tuple of the port used by credis and a handle to the process that was
+            started. If a port is passed in, then the returned port value is
+            the same.
+
+    Raises:
+        Exception: An exception is raised if credis could not be started.
+    """
+    redis_filepath = os.path.join(
```

Comment:
Please refactor most of this function body, so that we share it with the existing start_redis_instance().